File: Krestiki-noliki.c
#include <stdio.h>
#include <windows.h>
#include <stdlib.h>

void draw_table(char board[]) {
    printf(" %c | %c | %c \n", board[0], board[1], board[2]);
    printf("---+---+---\n");
    printf(" %c | %c | %c \n", board[3], board[4], board[5]);
    printf("---+---+---\n");
    printf(" %c | %c | %c \n", board[6], board[7], board[8]);
    
}

int check_win(char board[]) {  
if (board[0] == board[1] && board[1] == board[2]) return 1;
if (board[3] == board[4] && board[4] == board[5]) return 1;
if (board[6] == board[7] && board[7] == board[8]) return 1;
if (board[0] == board[3] && board[3] == board[6]) return 1;
if (board[1] == board[4] && board[4] == board[7]) return 1;
if (board[2] == board[5] && board[5] == board[8]) return 1;
if (board[0] == board[4] && board[4] == board[8]) return 1;
if (board[2] == board[4] && board[4] == board[6]) return 1;

    return 0;
}

int main(void) {
    SetConsoleCP(65001);
    SetConsoleOutputCP(65001);
    char boar[9] = {'1', '2', '3', '4', '5', '6', '7', '8', '9'};
    char name1[50];
    char name2[50];
    int hod=0,choice,flag=0;
    printf("Игра крестики-нолики\n");
    printf("Введите имя 1 игрока(X): ");
    scanf("%s",name1);
    printf("Введите имя 2 игрока(O): ");
    scanf("%s",name2);
    draw_table(boar);
    printf("\nИгрок 1 — %s\nИгрок 2 — %s\n", name1, name2);
    while (hod<9){
        if (hod%2==0){
            printf("Ход %s: ",name1);
            scanf("%d",&choice);
            int index=choice-1;
            if(choice>=1 && choice<=9 && boar[index]!='X' && boar[index]!='O'){
                boar[index]='X';
                hod++;
                system("cls");
                draw_table(boar);
                if (check_win(boar)==1){
                    printf("Победил %s!\n",name1);
                    flag+=1;
                    break;
                }
            }else{
                printf("Клетка занята или введена не правильно!\n");
                continue;
            }
        }else{
            printf("Ход %s: ",name2);
            scanf("%d",&choice);
            int ind=choice-1;
            if(choice>=1 && choice<=9 && boar[ind]!='X' && boar[ind]!='O'){
                boar[ind]='O';
                hod++;
                system("cls");
                draw_table(boar);
                if (check_win(boar)==1){
                    printf("Победил %s!\n",name2);
                    flag+=1;
                    break;
                }
            }else{
                printf("Клетка занята или введена не правильно!\n");
                continue;
        }
    }
    }
    if (flag==0){printf("Ничья\n");}
    return 0;
}

File: krestiki-noliki.py
print("   Крестики-нолик")
print("1 игорк-Х, 2 игок-О")

print(f"    1   |    2   |    3   ")
print("---------------------------")
print(f"    4   |    5   |    6   ")
print("---------------------------")
print(f"    7   |    8   |    9   ")
hod=0
player1=input("Имя 1-го игрока: ")
player2=input("Имя 2-го игрока: ")
odin=0
dva=0
tri=0
ch=0
pt=0
st=0
sm=0
vm=0
dv=0
c1=" "
c2=" "
c3=" "
c4=" "
c5=" "
c6=" "
c7=" "
c8=" "
c9=" "
def check_win():
    global c1,c2,c3,c4,c5,c6,c7,c8,c9
    if(c1==c2==c3!=" "):return True
    if(c4==c5==c6!=" "):return True
    if(c7==c8==c9!=" "):return True
    if(c1==c5==c9!=" "):return True
    if(c3==c5==c7!=" "):return True
    if(c1==c4==c7!=" "):return True
    if(c2==c5==c8!=" "):return True
    if(c3==c6==c9!=" "):return True
    return False
def draw_table():
    print(f"  {c1}  |  {c2}  |  {c3}  ")
    print("-----+-----+-----")
    print(f"  {c4}  |  {c5}  |  {c6}  ")
    print("-----+-----+-----")
    print(f"  {c7}  |  {c8}  |  {c9}  ")
try:
    while hod!=9:
        if hod%2==0:
            try:
                choice=int(input(f"{player1}: "))
            except ValueError:
                print("Цифры, дебил!")
                continue
            if choice<1 or choice>9:
                print("Вводить цифры от 1 до 9, дебил")
                continue
            
            elif choice==1:
                if odin!=1:
                    c1="X"
                    odin+=1
                else:
                    print(f"{player1}, эта клетка занятя, выберите другую")
                    continue
            elif choice==2:
                if dva!=1:
                    c2="X"
                    dva+=1
                else:
                    print(f"{player1}, эта клетка занятя, выберите другую")
                    continue
            elif choice==3:
                if tri!=1:
                    c3="X"
                    tri+=1
                else:
                    print(f"{player1}, эта клетка занятя, выберите другую")
                    continue
            elif choice==4:
                if ch!=1:
                    c4="X"
                    ch+=1
                else:
                    print(f"{player1}, эта клетка занятя, выберите другую")
                    continue
            elif choice==5:
                if pt!=1:
                    c5="X"
                    pt+=1
                else:
                    print(f"{player1}, эта клетка занятя, выберите другую")
                    continue

            elif choice==6:
                if st!=1:
                    c6="X"
                    st+=1
                else:
                    print(f"{player1}, эта клетка занятя, выберите другую")
                    continue
            elif choice==7:
                if sm!=1:
                    c7="X"
                    sm+=1
                else:
                    print(f"{player1}, эта клетка занятя, выберите другую")
                    continue
            elif choice==8:
                if vm!=1:
                    c8="X"
                    vm+=1
                else:
                    print(f"{player1}, эта клетка занятя, выберите другую")
                    continue
            elif choice==9:
                if dv!=1:
                    c9="X"
                    dv+=1
                else:
                    print(f"{player1}, эта клетка занятя, выберите другую")
                    continue
            draw_table()  
            if check_win():
                print(f"Победил {player1}")
                break
            hod+=1  
            
        else:
            try:
                choice=int(input(f"{player1}: "))
            except ValueError:
                print("Цифры, дебил!")
                continue
            if choice<1 or choice>9:
                print("Вводить цифры от 1 до 9, дебил")
                continue
            
            elif choice==1:
                if odin!=1:
                    c1="O"
                    odin+=1
                else:
                    print(f"{player2}, эта клетка занятя, выберите другую")
                    continue

            elif choice==2:
                if dva!=1:
                    c2="O"
                    dva+=1
                else:
                    print(f"{player2}, эта клетка занятя, выберите другую")
                    continue
            elif choice==3:
                if tri!=1:

                    c3="O"
                    tri+=1
                else:
                    print(f"{player2}, эта клетка занятя, выберите другую")
                    continue
            elif choice==4:
                if ch!=1:
                    c4="O"
                    ch+=1
                else:
                    print(f"{player2}, эта клетка занятя, выберите другую")
                    continue
            elif choice==5:
                if pt!=1:
                    c5="O"
                    pt+=1
                else:
                    print(f"{player2}, эта клетка занятя, выберите другую")
                    continue

            elif choice==6:
                if st!=1:
                    c6="O"
                    st+=1
                else:
                    print(f"{player2}, эта клетка занятя, выберите другую")
                    continue
            elif choice==7:
                if sm!=1:
                    c7="O"
                    sm+=1
                else:
                    print(f"{player2}, эта клетка занятя, выберите другую")
                    continue
            elif choice==8:
                if vm!=1:
                    c8="O"
                    vm+=1
                else:
                    print(f"{player2}, эта клетка занятя, выберите другую")
                    continue
            elif choice==9:
                if dv!=1:
                    c9="O"
                    dv+=1
                else:
                    print(f"{player2}, эта клетка занятя, выберите другую")
                    continue
            draw_table()
            if check_win():
                print(f"Победил {player2}")
                break
            
            hod+=1
        
except KeyboardInterrupt:
    print("\nЗавершение.")
